packages/EasyApiPlatform/src/Filter/AdvancedSearchFilter.php

```diff
@@ -0,0 +1,485 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyApiPlatform\Filter;
+
+use ApiPlatform\Api\IriConverterInterface;
+use ApiPlatform\Doctrine\Common\Filter\SearchFilterInterface;
+use ApiPlatform\Doctrine\Common\PropertyHelperTrait;
+use ApiPlatform\Doctrine\Orm\Filter\AbstractFilter;
+use ApiPlatform\Doctrine\Orm\Util\QueryBuilderHelper;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Exception\InvalidArgumentException;
+use ApiPlatform\Metadata\Operation;
+use Closure;
+use DateTimeInterface;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+/**
+ * Filter the collection by given properties.
+ * This filter allows to define multiple search strategy for the same property on a single resource.
+ *
+ * #[ApiFilter(
+ *     AdvancedSearchFilter::class,
+ *     properties: [
+ *         'email' => 'iexact', // Normal search property definition
+ *         'description[exact]' => ['description' => 'exact'], // Filter for description field with exact strategy
+ *         'description[partial]' => ['description' => 'partial'], // Filter for description field with partial strategy
+ *     ]
+ * )]
+ */
+final class AdvancedSearchFilter extends AbstractFilter implements SearchFilterInterface
```

Comment:
Can't we use `\ApiPlatform\Core\Bridge\Doctrine\Common\Filter\SearchFilterTrait` instead of copy-pasting of some methods?

Reply:
Thanks. Done.